Source/WebCore/inspector/InspectorStyleSheet.cpp

```diff
@@ -350,12 +365,34 @@ void StyleSheetHandler::endRuleBody(unsigned offset)
     m_currentRuleDataStack.last()->ruleBodyRange.end = offset;
     auto rule = popRuleData();
     fixUnparsedPropertyRanges(rule.ptr());
+
+    if (rule->containsImplicitlyNestedProperties&& rule->styleSourceData->propertyData.size()) {
+        // Property declarations made outside of a Style rule will become part of a special inferred `& {}` rule after
+        // the parser has already sent observers their notification of the property. In order to match the CSSOM
+        // representation this must be handled here as well, otherwise the flat rule lists will differ in length.
+        auto inferredStyleRule = CSSRuleSourceData::create(StyleRuleType::Style);
+        inferredStyleRule->isImplicitlyNested = true;
+        inferredStyleRule->ruleHeaderRange = rule->ruleHeaderRange;
+        inferredStyleRule->ruleBodyRange = rule->ruleBodyRange;
+
+        std::swap(rule->styleSourceData, inferredStyleRule->styleSourceData);
+
+        // Inferred style rules are always logically placed at the start of their parent rule.
+        rule->childRules.insert(0, WTFMove(inferredStyleRule));
```

Comment:
are there WPT for this that we previously would've failed but now pass?

Reply:
WPT does not test inspector behavior. We previously would end up with a mismatch between the CSSOM and our flattened source data vector, which meant that Web Inspector would show the incorrect rules for these, and likely for styles that followed in the style sheet as well.